kanidmd/src/lib/core/https/mod.rs

```diff
@@ -1161,6 +262,139 @@ impl<State: Clone + Send + Sync + 'static> tide::listener::Listener<State> for T
 }
 */
 
+struct NoCacheMiddleware;
+
+impl Default for NoCacheMiddleware {
+    fn default() -> Self {
+        NoCacheMiddleware {}
+    }
+}
+
+#[async_trait::async_trait]
+impl<State: Clone + Send + Sync + 'static> tide::Middleware<State> for NoCacheMiddleware {
+    async fn handle(
+        &self,
+        request: tide::Request<State>,
+        next: tide::Next<'_, State>,
+    ) -> tide::Result {
+        let mut response = next.run(request).await;
+        response.insert_header("Cache-Control", "no-store, max-age=0");
+        response.insert_header("Pragma", "no-cache");
+        Ok(response)
+    }
+}
+
+struct CacheableMiddleware;
+
+impl Default for CacheableMiddleware {
+    fn default() -> Self {
+        CacheableMiddleware {}
+    }
+}
+
+#[async_trait::async_trait]
+impl<State: Clone + Send + Sync + 'static> tide::Middleware<State> for CacheableMiddleware {
+    async fn handle(
+        &self,
+        request: tide::Request<State>,
+        next: tide::Next<'_, State>,
+    ) -> tide::Result {
+        let mut response = next.run(request).await;
+        response.insert_header("Cache-Control", "max-age=60,must-revalidate,private");
```

Comment:
What does this line do and what/how is it caching?

Reply:
It tells the client to cache content for a maximum of 60 seconds